docs/design-proposal.md

```diff
@@ -0,0 +1,190 @@
+and server CLI applications, unit tests, documentation, and build files. It 
+provides the ability to upload archive root file systems and to execute jobs
+consisting of commands or scripts located on the uploaded file systems. For this
+release, file system archives will be gzipped tarballs.
+
+Executing jobs cannot see the file system of the host the service is running on,
+or any of the other uploaded file systems other than the one its being executed
+on. Executing jobs have cannot access the host's network interfaces and have
+no network access of their own in this release.
+
+### IJR Library
+
+The library will be a Go package with all the code and APIs for job lifecycle 
+management. Documentation for the library can currently be found at in the 
+[docs/godocs](./godocs) and can be viewed by running the `make godocs` target 
+and navigating in a web browser to http://localhost:8000. Documentation is 
+currently statically generated files. This static documentation will be removed
+once the library is merged and documentation will be served by the `godocs`
+too.
+
+#### Technology Stack
+
+The library and its CLI apps will be implemented with the following technologies:
+
+* Golang 1.22 or later
+* Make for builds (see the [Makefile](../Makefile) for build targets)
+* gRPC for client/server transport
+* mTLS for client/server encryption
+
+The TLS 1.3 cipher suites that will be used include:
+
+* tls.TLS_AES_128_GCM_SHA256
+* tls.TLS_AES_256_GCM_SHA384
+* tls.TLS_CHACHA20_POLY1305_SHA256
+
+`tls.TLS_AES_128_GCM_SHA256` and `tls.TLS_AES_256_GCM_SHA384` both use the AES 
+cipher in Galois/Counter Mode (GCM) for symmetric encryption. AES is a widely 
+used symmetric encryption algorithm that is considered safe and efficient. 128 
+and 256 represent key sizes, where a larger key size generally means more 
+security but at the cost of more computational power. The numbers in the 
+SHA256 and SHA384 are the lengths of hashes in bits. SHA-256 and SHA-384 are 
+examples of hash functions in the SHA-2 family and are widely used in 
+cryptographic applications and protocols. They have longer hash lengths 
+compared to SHA-1 which is considered insecure now.
+
+`tls.TLS_CHACHA20_POLY1305_SHA256` uses the ChaCha20 stream cipher and Poly1305
+for message authentication. Stream ciphers are an alternative to block ciphers
+like AES. ChaCha20 is a newer stream cipher that is designed to offer improved
+performance on devices without hardware AES acceleration, such as some mobile
+devices. Poly1305 is used for ensuring message integrity. A fundamental
+component of secure communications, message authentication codes (MACs) like
+Poly1305 help assure that a message has not been tampered with in transit.
+
+The use of these cipher suites in TLS 1.3 implies encryption which offers 
+confidentiality along with integrity and authenticity, which are crucial for 
+secure communication. 
+
+### ijrd
```

Comment:
How are users of this server authorized?

Reply:
I missed that. I'll all a basic auth system when I repost the MR.